File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/NamingSystem.ts
import { UsageContext } from "./UsageContext";
import { Period } from "./Period";
import { ContactDetail } from "./ContactDetail";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { BackboneElement } from "./BackboneElement";

export type NamingSystemUniqueId = BackboneElement & {
    type: string;
    value: string;
    period?: Period;
    comment?: string;
    preferred?: boolean;
};

export type NamingSystem = DomainResource & {
    description?: string;
    date: string;
    publisher?: string;
    jurisdiction?: CodeableConcept[];
    name: string;
    useContext?: UsageContext[];
    type?: CodeableConcept;
    responsible?: string;
    usage?: string;
    status: string;
    kind: string;
    uniqueId: NamingSystemUniqueId[];
    contact?: ContactDetail[];
};

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/NamingSystem.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class NamingSystem : DomainResource, IResource
{
    public string? Description { get; set; }
    public required string Date { get; set; }
    public string? Publisher { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public required string Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public Base.CodeableConcept? Type { get; set; }
    public string? Responsible { get; set; }
    public string? Usage { get; set; }
    public required string Status { get; set; }
    public required string Kind { get; set; }
    public required NamingSystem_UniqueId[] UniqueId { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }

    public class NamingSystem_UniqueId : BackboneElement
    {
        public required string Type { get; set; }
        public required string Value { get; set; }
        public Base.Period? Period { get; set; }
        public string? Comment { get; set; }
        public bool? Preferred { get; set; }
    }
}